src/test/java/de/tum/in/www1/artemis/util/CourseTestService.java

```diff
@@ -1207,23 +1197,36 @@ public void testDownloadCourseArchiveAsInstructor_not_found() throws Exception {
 
     // Test
     public void testDownloadCourseArchiveAsInstructor() throws Exception {
+        submissionRepo.deleteAll();
 
-        // Dummy course archive
-        Path courseArchivePath = Path.of(courseArchivesDirPath, "some-course-archive.zip");
-        if (!Files.exists(courseArchivePath)) {
-            Files.createDirectories(Path.of(courseArchivesDirPath));
-            Files.createFile(courseArchivePath);
-        }
+        // Archive the course and wait until it's complete
+        testArchiveCourseWithTestModelingAndFileUploadExercises();
 
-        // Generate a course that has an archive
-        Course course = database.createCourse();
-        course.setCourseArchivePath(courseArchivePath.toString());
-        course = courseRepo.save(course);
+        var course = courseRepo.findAll().stream().findFirst().get();
+
+        // Download the archive
+        var archive = request.getFile("/api/courses/" + course.getId() + "/download-archive", HttpStatus.OK, new LinkedMultiValueMap<>());
+        assertThat(archive).isNotNull();
+        assertThat(archive).exists();
+
+        // Extract the archive
+        zipFileService.extractZipFileRecursively(archive.getAbsolutePath());
+        String extractedArchiveDir = archive.getPath().substring(0, archive.getPath().length() - 4);
```

Comment:
Do we need to check if the path length is >= 4? I'm not sure what substring does when given a negative parameter. (Probably throw an exception?)

Reply:
I think it's okay because filename of the archive is generated using a timestamp, id, and the name of the course/exercise/exam.